src/solver/Solver.java

```diff
@@ -54,9 +51,14 @@ public Board solve() {
 
             Vector<GameState> newGeneratedStates = analyzedBoard.nextStep();
             for (int i = 0; i < newGeneratedStates.size(); ++i) {
-                queue.add(newGeneratedStates.elementAt(i));
-                System.out.println("" + analyzed.isSolved() + " " + newGeneratedStates.elementAt(i).getBoardShortcut());
+                if(!mapsAddedToQueue.contains(newGeneratedStates.elementAt(i).getBoardShortcut())){
```

Comment:
Maybe will be better change iterating like `for (GameState state : newGeneratedStates)`, we don't use index in any other way that to get element from vector.